synapseclient/core/retry.py

```diff
@@ -340,7 +341,130 @@ async def foo(a, b, c): return [a, b, c]
                 (
                     "Retries have run out. re-raising the exception: %s"
                     if retry
-                    else "Rasing the exception: %s"
+                    else "Raising the exception: %s"
+                ),
+                str(caught_exception_info[0]),
+            )
+            raise caught_exception
+        return response
+
+
+def with_retry_time_based(
```

Comment:
Nearly a 1-to-1 copy from above. But without async/await. And using the time library to wait.